### PR TITLE
Fix incorrect version on resource parameter

### LINKED ISSUE #5812: Stu3 resource parameter for PlanDefinition/$apply is incorrect version

### PR DESCRIPTION
**Pull request by barhodes**
Closes #5812 

In addition to resolving the issue in 5812, several tests that were not being run have been fixed.

### CODE REVIEW

File: hapi-fhir-storage-cr/src/main/java/ca/uhn/fhir/cr/dstu3/plandefinition/PlanDefinitionApplyProvider.java [hunk 1]
```diff
@@ -88,7 +88,7 @@ public class PlanDefinitionApplyProvider {
 	@Operation(name = ProviderConstants.CR_OPERATION_APPLY, idempotent = true, type = PlanDefinition.class)
 	public IBaseResource apply(
 			@IdParam IdType theId,
-			@OperationParam(name = "planDefinition") org.hl7.fhir.r4.model.PlanDefinition thePlanDefinition,
+			@OperationParam(name = "planDefinition") PlanDefinition thePlanDefinition,
```
Comment (JPercival) [inline hapi-fhir-storage-cr/src/main/java/ca/uhn/fhir/cr/dstu3/plandefinition/PlanDefinitionApplyProvider.java:91]: question: Is it possible to add some tests for registering these providers to ensure we don't get mixups on versions again?

**Review by JPercival — APPROVED** [review]
question: do we need a changelog? This is essentially a fix for the previous integration.

File: hapi-fhir-jpaserver-test-utilities/src/main/java/ca/uhn/fhir/jpa/test/config/TestDstu3Config.java [hunk 2]
```diff
@@ -175,13 +175,6 @@ public DataSource dataSource() {
 		return dataSource;
 	}
 
-	@Bean
-	public IEmailSender emailSender() {
-		final MailConfig mailConfig = new MailConfig().setSmtpHostname("localhost").setSmtpPort(3025);
-		final IMailSvc mailSvc = new MailSvc(mailConfig);
-		return new EmailSenderImpl(mailSvc);
-	}
-
```
Comment (barhodes) [inline hapi-fhir-jpaserver-test-utilities/src/main/java/ca/uhn/fhir/jpa/test/config/TestDstu3Config.java:184]: This bean was failing during creating due to a Javax class not being found.  It does not appear to be used at all, removing it did not cause any other tests to fail and we can now use this in our CR Dstu3 tests.

**Comment by JPercival** [comment]
@barhodes - CR version 3.3.1 is released with those changes.

**Review by lukedegruchy — APPROVED** [review]

**Pull request merged**